### DISCUSSION TITLE
Observations, announcements, about the Boost Serialization library

### DISCUSSION THREAD

**Discussion by robertramey** [boostorg/serialization#306] [General]
This is a place for information regarding the Boost Serialization Library which doesn't fit anywhere else.  Things like:

- Issues which raise interesting subjects which cannot be immediately addressed and I'm loath to discard.
- suggestions for improving library

Basically stuff that would be in issues but that cannot be addressed for some reason or another.

**Comment by cpierre67** [comment]
Hello,
 
I have a program (main thread) launching a separate thread (of NORMAL priority) in which serialization in a boost stream is performed. At times, the serialization takes a long time (tens of seconds instead of usual tens of milliseconds) and hangs the main thread for that duration.
 
SERIALIZATION CULPRIT LINES: After positioning the 'cursor' in the boost stream std::fstream* fileos (linked to a file) using fileos->seekp, I perform serialization of the content of pItem (pointer to a class containing some FullVector and FullMatrix) using
          boost::archive::binary_oarchive oa(*(fileos));
          oa << *pItem;
These are the lines that hold the time.
 
The serialize function is just made of lines such as
                              ar & v1;
                              ar & v2;
                              ar & v3;
with ar of type Archive and v1,v2,v3 of type FullVector.
 
QUESTIONS:
- What can make the serialization take much longer time than usual? (Note that 'open' is performed earlier and 'flush' and 'close' are performed later.)
- Why does it hang the main thread (and maybe other resources)? Can it be prevented?

Thank you

**Reply by robertramey** [reply]
"I have a program (main thread) launching a separate thread (of NORMAL priority) in which serialization in a boost stream is performed. At times, the serialization takes a long time (tens of seconds instead of usual tens of milliseconds) and hangs the main thread for that duration."

I can't help but think that other parts of your program which might be running simultaneously with the serialization might be causing this.  Here are some suggestions:

1) Write a small test program to test just serialization and see if that exhibits the same symptoms 
2) Run the program with an execution time profiler.  

Hopefully this would yield some additional information which might be helpful in tracking down the source of the problem
